backfill.py

```diff
@@ -130,27 +148,126 @@ def get_result_from_durations(start_up_durations_path, analyzed_path):
+    commit_proc = subprocess.run(
+        ["git", "rev-list", "--ancestry-path", start_commit + "^.." + end_commit],
+        cwd=repository_path, capture_output=True, text=True)
+
+    if commit_proc.returncode != 0:
+        print(("\n\nSomething went wrong while checking out this commit range: {start}..{end}" +
+               "The associated error message was:\n\n {error}".format(
+                start=start_commit, end=end_commit, error=commit_proc.stderr.decode('utf-8').strip("\n"))),
+              file=sys.stderr)
+
+    return [e for e in commit_proc.stdout.split("\n") if e]
+
+
+def build_apk_for_commit(hash, repository_path, build_type):
+    checkout_proc = subprocess.run(["git", "checkout", hash], cwd=repository_path, capture_output=True)
+
+    if checkout_proc.returncode != 0:
+        print(("\n\nSomething went wrong while checking out this commit: {commit} . The associated error message was:"
+               "\n\n {error}".format(commit=hash, error=checkout_proc.stderr.decode('utf-8').strip("\n"))),
+              file=sys.stderr)
+
+    assemble_proc = subprocess.run(["./gradlew", "assemble"+build_type], cwd=repository_path, capture_output=True)
+
+    if assemble_proc.returncode != 0:
+        print(("\n\nSomething went wrong while assembling this build: {build} . The associated error message was:"
+               "\n\n {error}".format(build=build_type, error=checkout_proc.stderr.decode('utf-8').strip("\n"))),
+              file=sys.stderr)
+
+
+def build_apk_path_string(repository_path, build_type, phone_architecture):
+    apk_name = "app-{phone_arch}-{build_type}.apk".format(phone_arch=phone_architecture, build_type=build_type)
+    build_apk_destination = os.path.join(repository_path, "app", "build", "outputs", "apk", build_type, apk_name)
+    return build_apk_destination
+
+
+def move_apk_to_cwd(apk_path, commit_hash):
+    new_apk_name = "apk_commit_" + commit_hash + ".apk"
+    proc = subprocess.run(["mv", apk_path, new_apk_name])
+    if proc.returncode != 0:
+        print(("\n\nSomething went wrong while moving the built apk: {apk} . The associated error message was:"
+               "\n\n {error}".format(apk=apk_path, error=proc.stderr.decode('utf-8').strip("\n"))),
+              file=sys.stderr)
+    return new_apk_name
+
+
+def build_apks_for_commits(
+        start_commit=None, end_commit=None, repository_path=None,
+        build_type=None, architecture=None, remote_name=""):
+    apk_metadata_array = []
+
+    fetch_repository(repository_path, remote_name)
+    array_of_commit_hash = get_all_commits_in_commits_range(start_commit, end_commit, repository_path)
+    for commit in array_of_commit_hash:
+        build_apk_for_commit(commit, repository_path, build_type)
+        built_apk_name = build_apk_path_string(repository_path, build_type, architecture)
+        new_apk_name = move_apk_to_cwd(built_apk_name, commit)
+        apk_metadata_array.append({
```

Comment:
issue (non-blocking): if `build_apk_for_commit` fails, we can't run perf tests on that build but we're still signing ourselves up to do so.

suggestion: I think we need to `continue` if any of `build_apk_for_commit`, `move_apk_to_cwd` fails